matrix-room.c

```diff
@@ -793,6 +963,37 @@ void matrix_room_send_message(PurpleConversation *conv, const gchar *message)
     PurpleConvChat *chat = PURPLE_CONV_CHAT(conv);
     const char *type_string = "m.text";
     const gchar *message_to_send = message;
+    const char *image_start, *image_end;
+    GData *image_attribs;
+
+    /* Matrix doesn't have messages that have both images and text in, so
+     * we have to split this message if it has an image.
+     */
+    if (purple_markup_find_tag("img", message,
```

Comment:
Right, all handled nicely via a little recursion (and just tested as well - works :-)